.github/workflows/publish-preview.yml

```diff
@@ -0,0 +1,136 @@
+        
+      - name: Cache nuget
+        uses: actions/cache@v4
+        with:
+          path: ${{ github.workspace }}/.nuget/packages
+          key: ${{ runner.os }}-nuget-${{ github.sha }}
+          restore-keys: ${{ runner.os }}-nuget-
+        
+      - name: Cache agent
+        id: cache-agent
+        uses: actions/cache@v4
+        with:
+          path: src/Cody.VisualStudio/Agent
+          key: ${{ runner.os }}-agent-${{ hashFiles('agent/agent.version') }}  
+        
+      - name: Build agent if needed
+        if: ${{ steps.cache-agent.outputs.cache-hit != 'true' }}
+        shell: pwsh
+        run: ./agent/runBuildAgent.ps1
+        
+      - name: Set version on AssemblyInfo.cs files
+        uses: dannevesdantas/set-version-assemblyinfo@v.1.0.0
+        with:
+          version: ${{ steps.version.outputs.full_without_prefix }}
+          path: src/
+          
+      - name: Set version for .vsixmanifest file
+        uses: cezarypiatek/VsixVersionAction@1.2
+        with:
+          version: ${{ steps.version.outputs.full_without_prefix }}
+          vsix-manifest-file: src\Cody.VisualStudio\source.extension.vsixmanifest
+        
+      - name: Build extension (${{ env.Configuration }})
+        run: msbuild src/Cody.sln -t:Build -restore -verbosity:minimal -property:Configuration=${{ env.Configuration }}
+        
+      - name: Upload Cody.VisualStudio.vsix artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: Cody.VisualStudio.vsix
+          path: src/Cody.VisualStudio/bin/${{ env.Configuration }}/Cody.VisualStudio.vsix
+        
+      #Running tests  
+
+      - name: Change Screen Resolution
+        shell: pwsh
+        run: Set-DisplayResolution -Width 1920 -Height 1080 -Force
+        
+      - name: Run tests
+        env:
+          Access_Token_UI_Tests: ${{ secrets.SRC_ACCESS_TOKEN_DOTCOM }}
+          WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS: '--remote-debugging-port=9222'
+        run: dotnet vstest src/*Tests/bin/${{ env.Configuration }}/*.Tests.dll --logger:trx
+        
+      - name: Upload screenshots for UI tests
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: UI Tests Screenshots
+          path: src/Cody.VisualStudio.Tests/bin/${{ env.Configuration }}/Screenshots
+          retention-days: 10
+
+      - name: Publish Test Results
+        uses: EnricoMi/publish-unit-test-result-action/windows@v2
+        if: always()
+        with:
+          files: TestResults/**/*.trx
+          
+      #Publish
+          
+      - name: Create GitHub release
+        uses: ncipollo/release-action@v1.14.0
+        with:
+          tag: ${{ steps.version.outputs.full }}
+          name: Cody for Visual Studio ${{ steps.version.outputs.full_without_prefix }}
+          prerelease: true
+          artifacts: src/Cody.VisualStudio/bin/${{ env.Configuration }}/Cody.VisualStudio.vsix
+          artifactContentType: application/zip
```

Comment:
Could we turn off zipping `.vsix` installer?

Reply:
The installer is not zipped. It is just an additional piece of information that the blob of bytes being transferred is a zip archive.

Reply:
Could be this turned off? It's pretty annoying during testing that it's is not clean `.visx` file.

Reply:
Looks like it's not supported https://github.com/actions/upload-artifact/issues/426 😥